setup/kubernetes/openshift.md

```diff
@@ -0,0 +1,167 @@
+---
+title: "Red Hat OpenShift"
+description: Learn about deploying Coder in OpenShift Container Platform
+---
+
+This deployment guide shows you how to customize your [OpenShift Container
+Platform] cluster in order to deploy Coder. The OpenShift Container Platform
+includes default security features, notably the `restricted` [Security Context
+Constraint] (SCC), which can interfere with applications, including Coder.
+
+This guide describes customizations to the OpenShift cluster as well as Coder
+that ensure an optimal user experience.
+
+[OpenShift Container Platform]: https://www.openshift.com/products/container-platform
+[Security Context Constraint]: https://docs.openshift.com/container-platform/4.7/authentication/managing-security-context-constraints.html
+
+## Prerequisites
+
+- An OpenShift cluster with a Project (Kubernetes namespace) for Coder
+- OpenShift command-line tools (`oc` and `kubectl`)
```

Comment:
Should we add Postgres here? Do we need to include docs on how to spin up postgres on openshift?

Reply:
This is a great point -- Our other guides don't mention PostgreSQL, I think we just expect users to find their own :thinking:

Reply:
I am ok with that personally, if anyone ends up needing those docs, I'm sure we will hear about it. 

Maybe we should mention it in the prereqs though?

Reply:
Well, if we mention it for OpenShift, we should mention it for Azure/EKS/GKE, too, for consistency... and then I suppose it's best to do that as a separate PR